core/modules/parser/QSMySqlListener.cc

```diff
@@ -672,8 +690,17 @@ class RootAdapter :
         _selectStatement = selectStatement;
     }
 
+    void checkContext() const override {
+        // required:
+        assertExecutionCondition(__FUNCTION__, _ctx->EOF() != nullptr,
+                "Missing context condition: EOF is null.", _ctx);
+        // optional:
+        // MINUSMINUS()
+    }
+
     void onEnter(QSMySqlParser::RootContext* ctx, QSMySqlListener const * const listener) {
```

Comment:
Why the second `const` before `listener`?

Reply:
const pointer to a const object

Reply:
I'd argue that eliminating the second `const` will not change anything in the application.  The only difference only difference would be that the second `const` disallows modifying the variable within the body of the function. This change won't have any side effects beyond the method.
However, things would look totally different for:
```
void onEnter(...QSMySqlListener const ** const listener) 
```
In this case you do want to have the second `const` to prevent replacing the original pointer's value with a new object assigned within the method.